File: TDDMicroExercises.ProposedSolution/Java/TurnTicketDispenser/src/main/java/tddmicroexercises/turnticketdispenser/TicketDispenser.java
package tddmicroexercises.turnticketdispenser;

public class TicketDispenser
{
    private ITurnNumberSequence turnNumberSequence;
    
    public TicketDispenser(){
    	this(TurnNumberSequence.turnNumberSequence);
    }

	public TicketDispenser(ITurnNumberSequence turnNumberSequence) {
		this.turnNumberSequence = turnNumberSequence;
	}

	public TurnTicket getTurnTicket()
    {
        int newTurnNumber = turnNumberSequence.getNextTurnNumber();
        TurnTicket newTurnTicket = new TurnTicket(newTurnNumber);

        return newTurnTicket;
    }
}

File: scala/src/main/scala/tddmicroexercises/turnticketdispenser/TicketDispenser.scala
package tddmicroexercises.turnticketdispenser


class TicketDispenser {
  def getTurnTicket() : TurnTicket = {
    var newTurnNumber = TurnNumberSequence.getNextTurnNumber()
    var newTurnTicket = new TurnTicket(newTurnNumber)

    return newTurnTicket
  }
}